src/main/java/org/jabref/gui/maintable/CellFactory.java

```diff
@@ -10,96 +10,98 @@
+        //icon.setToolTipText(relevanceViewModel.getLocalization());
+        TABLE_ICONS.put(SpecialField.RELEVANCE.getFieldName(), icon);
 
         SpecialFieldViewModel qualityViewModel = new SpecialFieldViewModel(SpecialField.QUALITY, undoManager);
-        label = qualityViewModel.getIcon().getGraphicNode();
-        //label.setToolTipText(qualityViewModel.getLocalization());
-        TABLE_ICONS.put(SpecialField.QUALITY.getFieldName(), label);
+        icon = qualityViewModel.getIcon();
+        //icon.setToolTipText(qualityViewModel.getLocalization());
+        TABLE_ICONS.put(SpecialField.QUALITY.getFieldName(), icon);
 
         // Ranking item in the menu uses one star
         SpecialFieldViewModel rankViewModel = new SpecialFieldViewModel(SpecialField.RANKING, undoManager);
-        label = rankViewModel.getIcon().getGraphicNode();
-        //label.setToolTipText(rankViewModel.getLocalization());
-        TABLE_ICONS.put(SpecialField.RANKING.getFieldName(), label);
+        icon = rankViewModel.getIcon();
+        //icon.setToolTipText(rankViewModel.getLocalization());
+        TABLE_ICONS.put(SpecialField.RANKING.getFieldName(), icon);
 
         // Priority icon used for the menu
         SpecialFieldViewModel priorityViewModel = new SpecialFieldViewModel(SpecialField.PRIORITY, undoManager);
-        label = priorityViewModel.getIcon().getGraphicNode();
-        //label.setToolTipText(priorityViewModel.getLocalization());
-        TABLE_ICONS.put(SpecialField.PRIORITY.getFieldName(), label);
+        icon = priorityViewModel.getIcon();
+        //icon.setToolTipText(priorityViewModel.getLocalization());
+        TABLE_ICONS.put(SpecialField.PRIORITY.getFieldName(), icon);
 
         // Read icon used for menu
         SpecialFieldViewModel readViewModel = new SpecialFieldViewModel(SpecialField.READ_STATUS, undoManager);
-        label = readViewModel.getIcon().getGraphicNode();
-        //label.setToolTipText(readViewModel.getLocalization());
-        TABLE_ICONS.put(SpecialField.READ_STATUS.getFieldName(), label);
+        icon = readViewModel.getIcon();
+        //icon.setToolTipText(readViewModel.getLocalization());
+        TABLE_ICONS.put(SpecialField.READ_STATUS.getFieldName(), icon);
 
         // Print icon used for menu
         SpecialFieldViewModel printedViewModel = new SpecialFieldViewModel(SpecialField.PRINTED, undoManager);
-        label = printedViewModel.getIcon().getGraphicNode();
-        //label.setToolTipText(printedViewModel.getLocalization());
-        TABLE_ICONS.put(SpecialField.PRINTED.getFieldName(), label);
+        icon = printedViewModel.getIcon();
+        //icon.setToolTipText(printedViewModel.getLocalization());
+        TABLE_ICONS.put(SpecialField.PRINTED.getFieldName(), icon);
     }
 
     public Node getTableIcon(String fieldType) {
-        Node label = TABLE_ICONS.get(fieldType);
-        if (label == null) {
+        JabRefIcon icon = TABLE_ICONS.get(fieldType);
```

Comment:
If I understand the code below correctly, then this method is hit quite often with an empty (or null) string. In these cases (checked using `StringUtil.isBlank`), we can simply return null before looking in the `Map` for a suitable icon.

Reply:
Done!